### ISSUE TITLE
jinja markdown extension compiles jinja in code blocks

### ISSUE THREAD

**Issue by colmjude** [digital-land/design-system#1]
The markdown compiler can't handle code blocks that contain jinja. E.g.

```
{% block bodyEnd %}
{{ super() }}
<script>(Some JS)</script>
{% endblock %}
```

An error is thrown. This is because the markdown-jinja extension sees it and tries to compile it. It should ignore anything in code blocks.

**Comment by colmjude** [comment]
I think before the markdown-jinja extension tries to compile any jinja it sees in the markdown file it needs to extract any code blocks. Then it can compile any legitimate jinja. Then just before handing the lines back to the markdown compile it needs to re-insert the code blocks in the right place(s).

**Comment by colmjude** [comment]
This has been fixed with commit 14e678c.

The fix works for fenced code-blocks only. So to avoid errors any jinja code blocks need to use the **```** syntax.
